### DISCUSSION TITLE
Sentinel 3 SL_2_LST via Iris

### DISCUSSION THREAD

**Discussion by Rapsodia86** [SciTools/iris#5135] [Q&A]
Hi All,
Wanted to ask if anyone used Iris to work with Land Surface Temperature product level 2 from Sentinel 3? (https://sentinels.copernicus.eu/documents/247904/1872792/Sentinel-3-SLSTR-Product-Data-Format-Specification-Level-2-Land.pdf/e587c02b-1454-46ff-b74e-bfc91943ae78?t=1637223909444)
This dataset is encoded in NetCDF 4; Lat and Long are in a separate *.nc file. 

Thanks!

**Comment by trexfeathers** [comment]
We're keen to hear from the user community on this.

But in principle this should work, with some manual intervention. Providing the Lat and Lon are the correct lengths for the dataset shape (which can't be guaranteed when separated into files), then you can use the [`Cube.add_dim_coord()`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/cube.html#iris.cube.Cube.add_dim_coord) method to add them to the dataset `Cube`.

Note that when loaded from a separate file, Lat and Lon might also have been interpreted as their own `Cube`s, but these would still contain all the information you would need to manually construct your own [`DimCoord`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/coords.html#iris.coords.DimCoord)s.

**Reply by Rapsodia86** [reply]
Ok, at some point will test the approach and report back! 
But also would be happy to hear from other users, at least maybe on a separate coord file.

Thanks!

**Reply by trexfeathers** [reply]
Hi @Rapsodia86, any news on this?

**Reply by Rapsodia86** [reply]
Hi there, not yet. I have been recently using R, and tried there: https://github.com/r-spatial/stars/issues/618 , but I am not 100% satisfied though may be getting closer. Those curvilinear grids are not nice to work with.
